packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -197,33 +198,102 @@ export function getRootHostContext(
   rootContainerInstance: Container,
 ): HostContext {
   let type;
-  let namespace: HostContextProd;
+  let context: HostContextProd;
   const nodeType = rootContainerInstance.nodeType;
   switch (nodeType) {
     case DOCUMENT_NODE:
     case DOCUMENT_FRAGMENT_NODE: {
       type = nodeType === DOCUMENT_NODE ? '#document' : '#fragment';
       const root = (rootContainerInstance: any).documentElement;
-      namespace = root ? root.namespaceURI : getChildNamespace(null, '');
+      if (root) {
+        const namespaceURI = root.namespaceURI;
+        context = namespaceURI
+          ? getOwnHostContext(namespaceURI)
+          : HostContextNamespaceNone;
+      } else {
+        context = HostContextNamespaceNone;
+      }
       break;
     }
     default: {
       const container: any =
         nodeType === COMMENT_NODE
           ? rootContainerInstance.parentNode
           : rootContainerInstance;
-      const ownNamespace = container.namespaceURI || null;
       type = container.tagName;
-      namespace = getChildNamespace(ownNamespace, type);
+      const namespaceURI = container.namespaceURI;
+      if (!namespaceURI) {
+        switch (type) {
+          case 'svg':
```

Comment:
don't these need to be uppercase?

Reply:
I copied it from here:

https://github.com/facebook/react/blob/e2816c3ffad96dfd45376e7a94e48012505b447b/packages/react-dom-bindings/src/client/DOMNamespaces.js#L16-L22

Reply:
SVG and MathML nodes are case-sensitive.

Reply:
🤯

Reply:
I knew the inner ones were but I always thought of svg and math themselves as html tags, just like foreignObject is an svg tag not an html tag